### DISCUSSION TITLE
Nuxt link changes the url in browser and setup runs, but the new page doesn't load

### DISCUSSION THREAD

**Discussion by chrissyast** [nuxt/nuxt#20177] [Questions]
```js
  //linking page
  <template>
    <nuxt-link :to="link">
       // a button
    </nuxt-link>
    ...
  </template>
  
  <script setup>
    ...
    const link = computed(() => `copy/${from.id}/${to.id}`)
    ...
  </script>
```
Clicking the button changes the url in the browser to `{baseUrl}/copy/foo/bar` as expected. The <RouterView> in devtools changes too.

```js
  // pages/copy/[from]/[to].vue
  <template>
    <div>
        Hello
    </div>
  </template>


  <script setup>
    const { data: success, pending } = await // API call 
  </script>
```

I know that this API call is happening because a request hits the backend, but the view stays on the current page. If I refresh the page I see "hello" as expected.

**Comment by chrissyast** [comment]
I noticed this also coincides with an error in the console

```
unk-3NMN3MUW.js:7946 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'parentNode')
    at remove (chunk-3NMN3MUW.js:7946:1)
    at Object.remove (chunk-3NMN3MUW.js:6812:1)
    at unmount (chunk-3NMN3MUW.js:6488:1)
 
```

**Comment by warflash** [comment]
Hard to tell without reproduction but looks like you might have multiple root elements inside your `linking page` which can cause some issues.

**Reply by chrissyast** [reply]
I managed to track down the problem. It was a vuetify component which was causing hydration errors and also the error on unmount. https://github.com/vuetifyjs/vuetify/issues/15323

**Reply by warflash** [reply]
Ah nice, I see. 
Yeah I've run into lots of issues like this with vuetify in the past which is why I don't really use it anymore.
On that note, I'd recommend wrapping library components with your own component so you can easily swap them out - or in this case wrap the dialog in client-only so you only have to do that once :)

**Reply by chrissyast** [reply]
I used `v-no-ssr` which as far as I can tell is basically the same behaviour as `client-only`

**Comment by ndragun92** [comment]
@chrissyast always put / before the link that can cause an issues

instead of 
    ```const link = computed(() => `copy/${from.id}/${to.id}`)```
    
    write 
   ``` const link = computed(() => `/copy/${from.id}/${to.id}`)```

**Reply by chrissyast** [reply]
It was the above mentioned vuetify/hydration issue. On solving this the link worked fine, even without the added slash.
